File: app/src/main/java/com/pims/medretailers/SalesOrder.java
package com.pims.medretailers;

public class SalesOrder {
    private String id;              // "name" in API
    private String customerName;    // "customer_name" or "customer"
    private String ofsStatus;       // "delivery_status" (Order Fulfillment)
    private String approvalStatus;  // "status" or "workflow_state"
    private String date;            // "transaction_date"
    private String territory;       // "territory"
    private double grandTotal;      // "grand_total"
    private boolean isSelected;

    public SalesOrder(String id, String customerName, String ofsStatus, String approvalStatus, String date, String territory, double grandTotal) {
        this.id = id;
        this.customerName = customerName;
        this.ofsStatus = ofsStatus;
        this.approvalStatus = approvalStatus;
        this.date = date;
        this.territory = territory;
        this.grandTotal = grandTotal;
        this.isSelected = false;
    }

    // Getters
    public String getId() { return id; }
    public String getCustomerName() { return customerName; }
    public String getOfsStatus() { return ofsStatus; }
    public String getApprovalStatus() { return approvalStatus; }
    public String getDate() { return date; }
    public String getTerritory() { return territory; }
    public double getGrandTotal() { return grandTotal; }
    public boolean isSelected() { return isSelected; }
    public void setSelected(boolean selected) { isSelected = selected; }
}

File: lib/models/sales_order.dart
class SalesOrder {
  final String id;
  final String customerName;
  final String ofsStatus;
  final String approvalStatus;
  final String date;
  final String territory;
  final double grandTotal;
  bool isSelected;

  SalesOrder({
    required this.id,
    required this.customerName,
    required this.ofsStatus,
    required this.approvalStatus,
    required this.date,
    required this.territory,
    required this.grandTotal,
    this.isSelected = false,
  });

  factory SalesOrder.fromJson(Map<String, dynamic> json) {
    return SalesOrder(
      id: json['name'] ?? "N/A",
      customerName: json['customer_name'] ?? "Unknown Customer",
      ofsStatus: json['fulfillment_status'] ?? "N/A",
      approvalStatus: json['workflow_state'] ?? "Draft",
      date: json['transaction_date'] ?? "N/A",
      territory: json['territory'] ?? "N/A",
      grandTotal: (json['grand_total'] ?? 0.0).toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': id,
      'customer_name': customerName,
      'fulfillment_status': ofsStatus,
      'workflow_state': approvalStatus,
      'transaction_date': date,
      'territory': territory,
      'grand_total': grandTotal,
    };
  }
}
